src/cast_core/include/cast_core/actors/StreamStatsReporter.hpp

```diff
@@ -0,0 +1,80 @@
+// Copyright 2023-present MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef HEADER_D5EB1FAB_E817_43DE_A126_FCC6995B4C3C_INCLUDED
+#define HEADER_D5EB1FAB_E817_43DE_A126_FCC6995B4C3C_INCLUDED
+
+#include <string_view>
+
+#include <mongocxx/pool.hpp>
+
+#include <gennylib/Actor.hpp>
+#include <gennylib/PhaseLoop.hpp>
+#include <gennylib/context.hpp>
+
+#include <metrics/metrics.hpp>
+
+namespace genny::actor {
+
+/**
+ * Periodically polls (every 250ms) the stream processor's stats from the
```

Comment:
Do this every 1s or 2s instead? Stats polling is not super cheap for stream processor and adding another 2s should not affect the throughput as our throughput test should anyways be running for minutes.